storage/integration_test.go

```diff
@@ -3428,199 +3428,195 @@ func TestIntegration_CancelWrite(t *testing.T) {
-
-	initialSettings := []CORS{
-		{
-			MaxAge:          time.Hour,
-			Methods:         []string{"POST"},
-			Origins:         []string{"some-origin.com"},
-			ResponseHeaders: []string{"foo-bar"},
-		},
-	}
+	multiTransportTest(context.Background(), t, func(t *testing.T, ctx context.Context, _ string, prefix string, client *Client) {
+		initialSettings := []CORS{
+			{
+				MaxAge:          time.Hour,
+				Methods:         []string{"POST"},
+				Origins:         []string{"some-origin.com"},
+				ResponseHeaders: []string{"foo-bar"},
+			},
+		}
 
-	for _, test := range []struct {
-		input []CORS
-		want  []CORS
-	}{
-		{
-			input: []CORS{
-				{
-					MaxAge:          time.Hour,
-					Methods:         []string{"GET"},
-					Origins:         []string{"*"},
-					ResponseHeaders: []string{"some-header"},
+		for _, test := range []struct {
+			desc  string
+			input []CORS
+			want  []CORS
+		}{
+			{
+				desc: "set new values",
+				input: []CORS{
+					{
+						MaxAge:          time.Hour,
+						Methods:         []string{"GET"},
+						Origins:         []string{"*"},
+						ResponseHeaders: []string{"some-header"},
+					},
 				},
-			},
-			want: []CORS{
-				{
-					MaxAge:          time.Hour,
-					Methods:         []string{"GET"},
-					Origins:         []string{"*"},
-					ResponseHeaders: []string{"some-header"},
+				want: []CORS{
+					{
+						MaxAge:          time.Hour,
+						Methods:         []string{"GET"},
+						Origins:         []string{"*"},
+						ResponseHeaders: []string{"some-header"},
+					},
 				},
 			},
-		},
-		{
-			input: []CORS{},
-			want:  nil,
-		},
-		{
-			input: nil,
-			want: []CORS{
-				{
-					MaxAge:          time.Hour,
-					Methods:         []string{"POST"},
-					Origins:         []string{"some-origin.com"},
-					ResponseHeaders: []string{"foo-bar"},
+			{
+				desc:  "set to empty to remove existing policies",
+				input: []CORS{},
+				want:  nil,
+			},
+			{
+				desc:  "do not set to keep existing policies",
+				input: nil,
+				want: []CORS{
+					{
+						MaxAge:          time.Hour,
+						Methods:         []string{"POST"},
+						Origins:         []string{"some-origin.com"},
+						ResponseHeaders: []string{"foo-bar"},
+					},
 				},
 			},
-		},
-	} {
-		bkt := client.Bucket(uidSpace.New())
-		h.mustCreate(bkt, testutil.ProjID(), &BucketAttrs{CORS: initialSettings})
-		defer h.mustDeleteBucket(bkt)
-		h.mustUpdateBucket(bkt, BucketAttrsToUpdate{CORS: test.input}, h.mustBucketAttrs(bkt).MetaGeneration)
-		attrs := h.mustBucketAttrs(bkt)
-		if diff := testutil.Diff(attrs.CORS, test.want); diff != "" {
-			t.Errorf("input: %v\ngot=-, want=+:\n%s", test.input, diff)
+		} {
+			t.Run(test.desc, func(t *testing.T) {
+				h := testHelper{t}
+
+				bkt := client.Bucket(prefix + uidSpace.New())
+				h.mustCreate(bkt, testutil.ProjID(), &BucketAttrs{CORS: initialSettings})
+				defer h.mustDeleteBucket(bkt)
+				// Set VersioningEnabled attr so we don't fail when setting CORS to nil
```

Comment:
Maybe clarify that this is so we don't send an empty update/patch request, which is invalid for gRPC